libasherah_test.go

```diff
@@ -0,0 +1,730 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/godaddy/cobhan-go"
+)
+
+func validSetupForTesting(t *testing.T) {
+	config := AsherahConfig{}
+
+	config.KmsType = "static"
+	config.ServiceName = "TestService"
+	config.ProductID = "TestProduct"
+	config.Metastore = "memory"
+	config.SessionCache = true
+	config.SessionCacheDuration = 1000
+	config.SessionCacheMaxSize = 2
+	config.ExpireAfter = 1000
+	config.CheckInterval = 1000
+	config.DebugOutput = true
+	config.RegionMapStr = "region1=arn1,region2=arn2"
+
+	jsonBytes, err := json.Marshal(config)
+	if err != nil {
+		t.Error(err)
+	}
+
+	buf := cobhan.AllocateBuffer(1024)
+
+	result := cobhan.BytesToBufferSafe(jsonBytes, &buf)
+	if result != ERR_NONE {
+		t.Error(fmt.Sprintf("BytesToBufferSafe returned %v", result))
+	}
+
+	SetupJson(cobhan.Ptr(&buf))
+}
+
+func testAllocateStringBuffer(t *testing.T, str string) []byte {
+	buf, result := cobhan.AllocateStringBuffer(str)
+	if result != ERR_NONE {
+		t.Error(fmt.Sprintf("AllocateStringBuffer returned %v", result))
+		t.FailNow()
+	}
+	return buf
+}
+
+func TestSetupJson(t *testing.T) {
+	validSetupForTesting(t)
+	Shutdown()
+}
+
+func TestSetupJson2(t *testing.T) {
```

Comment:
Let's use a more descriptive name to make it obvious what we're testing in this function vs `TestSetupJson`